src/components/Payment/Payment.tsx

```diff
@@ -149,11 +154,16 @@ const Payment = ({
                     method: transaction.paymentMethod,
                   })}
                 </p>
-                <p>
-                  {transaction.transactionId}
-                  <br />
-                  {formatDate(transaction.transactionDate)}
-                </p>
+                <div className={styles.transactionDetails}>
+                  <p>
+                    {transaction.transactionId}
+                    <br />
+                    {formatDate(transaction.transactionDate)}
+                  </p>
+                  <IconButton aria-label={t('user:payment.show_receipt')} onClick={() => !isLoading && onShowReceiptClick(transaction.transactionId)}>
+                    <ExternalLink />
+                  </IconButton>
```

Comment:
Do you think we can see if we can show the invoice? For example, using a service feature flag?

I also remember that we should hide the button when the invoice price is zero/granted because these invoices don't work.